app/test_main.py

```diff
@@ -1 +1,26 @@
-# write your code here
+import unittest
+from unittest.mock import patch
+from datetime import datetime
+from app.main import can_access_google_page
+
+
+class TestGoogleFunctions(unittest.TestCase):
+    def setUp(self) -> None:
+        self.mock_datetime = patch("datetime.datetime").start()
+        self.mock_get = patch("requests.get").start()
```

Comment:
Instead of mocking the external modules (datetime and requests), you could  mock the specific functions (has_internet_connection and valid_google_url). Mocking the lower-level external module methods (like you did) will also work, but  mocking at the function level is a more direct way of controlling behavior since it focuses on your own functions rather than external modules.